pkgs/build-support/dotnet/make-nuget-deps/default.nix

```diff
@@ -1,14 +1,30 @@
-{ linkFarmFromDrvs, fetchurl }:
+{ linkFarmFromDrvs, substituteAll, fetchurl, stdenvNoCC, zip }:
 { name, nugetDeps ? import sourceFile, sourceFile ? null }:
 linkFarmFromDrvs "${name}-nuget-deps" (nugetDeps {
   fetchNuGet = { pname, version, sha256 ? "", hash ? ""
     , url ? "https://www.nuget.org/api/v2/package/${pname}/${version}" }:
-    fetchurl {
+    stdenvNoCC.mkDerivation {
       name = "${pname}.${version}.nupkg";
-      # There is no need to verify whether both sha256 and hash are
-      # valid here, because nuget-to-nix does not generate a deps.nix
-      # containing both.
-      inherit url sha256 hash;
+      src = fetchurl {
+        name = "${pname}.${version}.nupkg.${url}";
```

Comment:
do we need to set `name` here?  I would expect the default to be fine.

Reply:
I guess if I removed the `.${url}` from the end again, but kept the custom `name`, we could reuse whatever's already in people's stores from `fetchNuget` as of current master. But I don't have a strong opinion; I'm currently running a `nixpkgs-review` with the `name` omitted entirely.

Reply:
Omitting the `name` entirely appears to give us a pretty nasty name of "just the version number, nothing else":

> [10/788/8298 built, 1/324/325 copied (156.6/158.1 MiB), 111.6/111.8 MiB DL] building 17.0.0: calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/04w1mgkhawycg64s4q3n8mkcvma5ppx0-apple-framework-CoreFoundation-11.0.0

Reply:
Ah yeah, that's because the default source url is the v2 version which is `"https://www.nuget.org/api/v2/package/${pname}/${version}"` :(

I'm fine with keeping the explicit name for both the stripped and src ones.  one of them could have a suffix to differentiate them, but I don't think we should put the whole url in the name.

Reply:
I'm okay with what's in here now, but I was wondering about 

> I guess if I removed the .${url} from the end again, but kept the custom name, we could reuse whatever's already in people's stores from fetchNuget as of current master.

Maybe we should still do this, to save some downloads?

We'd just need to come up with a name for the stripped version I guess.  Does it need to end with .nupkg?  If so, maybe `"${pname}.${version}-stripped.nupkg"` or even `"stripped-${pname}.${version}.nupkg" if the end of the filename matters?

Reply:
It doesn't *have* to have a different name, I guess - they're distinguished by their input hashes, in the input-addressed store, or they are content-identical and deduplicated in the content-addressed store. They could both just be called `${pname}.${version}.nupkg`.

Reply:
Yeah, I wouldn't mind if they have the same names, if differentiating them causes some sort of problem.